examples/particles.rs

```diff
@@ -0,0 +1,309 @@
+                client.set_game_mode(GameMode::Creative);
+                client.teleport(
+                    [
+                        SPAWN_POS.x as f64 + 0.5,
+                        SPAWN_POS.y as f64 + 1.0,
+                        SPAWN_POS.z as f64 + 0.5,
+                    ],
+                    0.0,
+                    0.0,
+                );
+                client.set_player_list(server.state.player_list.clone());
+
+                if let Some(id) = &server.state.player_list {
+                    server.player_lists.get_mut(id).insert(
+                        client.uuid(),
+                        client.username(),
+                        client.textures().cloned(),
+                        client.game_mode(),
+                        0,
+                        None,
+                    );
+                }
+            }
+
+            if client.is_disconnected() {
+                self.player_count.fetch_sub(1, Ordering::SeqCst);
+                if let Some(id) = &server.state.player_list {
+                    server.player_lists.get_mut(id).remove(client.uuid());
+                }
+                server.entities.remove(client.state);
+
+                return false;
+            }
+
+            let entity = server
+                .entities
+                .get_mut(client.state)
+                .expect("missing player entity");
+
+            while handle_event_default(client, entity).is_some() {}
+
+            true
+        });
+
+        if !server.clients.is_empty() && server.shared.current_tick() % 30 == 0 {
+            if server.state.particle_index == server.state.particle_list.len() {
+                server.state.particle_index = 0;
+            }
+            let pos = Vec3::new(0.0, 100.0, -10.0);
+            let offset = Vec3::new(0.5, 0.5, 0.5);
+            let opt: Option<&ParticleType> =
+                server.state.particle_list.get(server.state.particle_index);
+            match opt {
+                None => {
+                    unreachable!("Invalid index to particle list");
+                }
+                Some(particle_type) => {
+                    println!("Current particle: {}", particle_type.name());
+                    server.clients.iter_mut().for_each(|(_cid, client)| {
+                        client.set_title(
+                            "",
+                            particle_type.name().to_string().bold(),
+                            SetTitleAnimationTimes {
+                                fade_in: 0,
+                                stay: 100,
+                                fade_out: 2,
+                            },
+                        );
+                        client.play_particle(particle_type.clone(), pos, offset, 0.1, 100, true);
+                    });
+                }
+            }
+            server.state.particle_index += 1;
+        }
+    }
+}
+
+fn add_all_particles(server: &mut Server<Game>) {
```

Comment:
Looks like you want to use the `vec!` macro in the `main` function to initialize the vector instead.